imports/plugins/core/discounts/client/components/form.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component, PropTypes } from "react";
+import { Meteor } from "meteor/meteor";
+import debounce from "lodash/debounce";
+import { Translation } from "/imports/plugins/core/ui/client/components";
+
+export default class DiscountForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      discount: this.props.discount,
+      validatedInput: false,
+      attempts: 0,
+      discountApplied: false
+    };
+
+    // debounce helper so to wait on user input
+    this.debounceDiscounts = debounce(() => {
+      const { discount } = this.state;
+      Meteor.call("discounts/codes/apply", this.props.id, discount, this.props.collection, (error, results) => {
+        if (results) {
+          this.setState({ discountApplied: results, discount: "" });
+        } else {
+          this.setState({ discountApplied: false });
+        }
+      });
+    }, 500);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  // handle apply
+  renderApplied() {
+    return (
+      <a onClick={this.handleClick}>
+        <Translation defaultValue="Discount submitted." i18nKey="discounts.submitted"/>
+      </a>
+    );
+  }
+
+  // handle keydown and change events
+  handleChange(event) {
+    const { attempts } = this.state;
+    // clear input if user hits escape key
+    if (event.keyCode === 27) {
```

Comment:
Might be worth handling the `enter` key as well so that the payment form doesn't submit if the user hits enter after inputting the code.